newscoop/classes/Section.php

```diff
@@ -173,21 +173,46 @@ public function copy($p_destPublicationId, $p_destIssueNumber,
      */
     public function delete($p_deleteArticles = false, $p_deleteArticleTranslations = false)
     {
-        $numArticlesDeleted = 0;
+        $articlesCount = 0;
         if ($p_deleteArticles) {
             $languageId = null;
             if (!$p_deleteArticleTranslations) {
                 $languageId = $this->m_data['IdLanguage'];
             }
-            $articles = Article::GetArticles($this->m_data['IdPublication'],
-                        $this->m_data['NrIssue'],
-                        $this->m_data['Number'],
-                        $languageId);
-            $numArticlesDeleted = count($articles);
-            foreach ($articles as $deleteMe) {
-                $deleteMe->delete();
+
+            $articlesCount = Article::GetArticles(
+                $this->m_data['IdPublication'],
+                $this->m_data['NrIssue'],
+                $this->m_data['Number'],
+                $languageId,
+                null,
+                true
+            );
+
+            $batch = 30;
+            $steps = ($articlesCount > $batch) ? ceil($articlesCount / $batch) : 1;
+
+            for ($i = 0; $i < $steps; $i++) {
```

Comment:
check line 211, it invokes `delete` on `\Article` object, where inside that method it removes all the associations to that article, geo maps, webcodes, images, topics, article data etc, this would then require a huge refactor I guess